pkg/login/social/connectors/generic_oauth.go

```diff
@@ -276,11 +286,14 @@ func (s *SocialGenericOAuth) UserInfo(ctx context.Context, client *http.Client,
 		}
 	}
 
-	if userInfo.Role == "" && !s.info.SkipOrgRoleSync {
-		if s.info.RoleAttributeStrict {
-			return nil, errRoleAttributeStrictViolation.Errorf("idP did not return a role attribute")
+	if !s.info.SkipOrgRoleSync {
+		userInfo.OrgRoles = s.orgRoleMapper.MapOrgRoles(s.orgMappingCfg, externalOrgs, userInfo.Role)
+		if s.info.RoleAttributeStrict && len(userInfo.OrgRoles) == 0 {
+			// If no roles are found and role_attribute_strict is set, return an error.
+			// The s.info.RoleAttributeStrict is necessary, because there is a case when len(userInfo.OrgRoles) == 0,
+			// but strict role mapping is not enabled (when getAllOrgs fails).
+			return nil, errRoleAttributeStrictViolation.Errorf("could not evaluate any valid roles using IdP provided data")
 		}
-		userInfo.Role = s.defaultRole()
```

Comment:
Do we not need to set a default if no roles are mapped through org role mapping but `RoleAttributeStrict` is disabled?

Reply:
I added a test (`""Give AutoAssignOrgRole in AutoAssignOrgId when OrgMapping returns no OrgRoles"` ) showing this is already the case, thanks to the OrgRoleMapper:

https://github.com/grafana/grafana/blob/7726631fe8f5220fe09f8854616993fc1465bd0e/pkg/login/social/connectors/org_role_mapper.go#L63-L65

Reply:
I see, thanks! I'll do a bit more manual testing for edge cases like this early next week, but I think overall the code looks good, so hopefully we can get this merged very soon 🤞